pkg/statefulset-grid-daemon/controller/controller.go

```diff
@@ -256,17 +293,30 @@ func (ssgdc *StatefulSetGridDaemonController) syncDnsHosts(key string) error {
 
 	set, err := ssgdc.setLister.StatefulSets(namespace).Get(name)
 	if errors.IsNotFound(err) {
-		klog.V(2).Infof("statefulset %v has been deleted", key)
+		klog.V(2).Infof("StatefulSet %v has been deleted", key)
 		return nil
 	}
 	if err != nil {
 		return err
 	}
 
+	var PodDomainInfoToHosts = make(map[string]string)
+	ControllerRef := metav1.GetControllerOf(set)
+	// Check existence of statefulset relevant service and execute delete operations if necessary
+	if needClear, err := ssgdc.needClearStatefulSetDomains(set); err != nil {
+		return err
+	} else if needClear {
+		if err := ssgdc.hosts.CheckOrUpdateHosts(PodDomainInfoToHosts, set.Namespace, ControllerRef.Name, set.Spec.ServiceName); err != nil {
+			klog.Errorf("Clear statefulset %v dns hosts err %v", key, err)
+			return err
+		}
+		klog.V(4).Infof("Clear statefulset %v dns hosts successfully", key)
+	}
```

Comment:
Return here? cause service and records has been deleted, process should be stoped?

Reply:
That's right, I have updated it already.